core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -86,6 +91,23 @@ public Long getDeadlineNanoTime() {
     return deadlineNanoTime;
   }
 
+  /**
+   * Retruns the Codec used for compression and decompression, or {@code null} if none is set.
```

Comment:
This makes it sound like the response must be using the same compression type if using compression. However, the server is permitted to choose a different compression algorithm if the client supports it. How is `Message-Accept-Encoding` going to play into this?